src/convertToObject.js

```diff
@@ -5,8 +5,26 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
+function convertToObject(stylesString) {
+  return stylesString
+    .split(';')
+    .map((style) => style.trim())
+    .filter(Boolean)
+    .map((style) => {
+      const [property, value] = style.split(':');
+
+      if (property === undefined || value === undefined) {
+        return null;
+      }
+
+      return [property.trim(), value.trim()];
+    })
+    .filter(Boolean)
+    .reduce((obj, [property, value]) => {
```

Comment:
[CODE KNOWLEDGE] - There's no need to filter the array again after the map on line 13. The previous filter on line 12 already ensures that all elements are truthy. The map will not introduce any new falsy values, so the filter on line 22 is redundant and can be removed.